### PR TITLE
Client student upload tff106

### PR DESCRIPTION
**Pull request by gearedmonkey**
### Summary
You can upload students.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Start server on parse-file-tff107
2. Start client
3. Click the upload students link on main page
4. Select file which is in docs/Mock_students.xlsx
5. Click submit
6. Verify toast appears for uploading students
7. Enter id of newly created student
8. Verify checkout page shows the new student

### Links

- [TFF-106](https://msoese.atlassian.net/browse/TFF-106)
- TheFourFifths/consus/pull/44

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,6 +65,7 @@
     "react-dom": "^15.4.1",
     "react-router": "^2.8.1",
     "request": "^2.75.0",
-    "sinon": "^1.17.6"
+    "sinon": "^1.17.6",
+    "react-simple-file-input": "latest"
```
Comment (thofmann) [inline package.json]: Why latest?

**Comment by mattmahn** [comment]
>### Summary
>
>Provide a general description of the code changes in your pull request... were
>there any bugs you had fixed? If so, mention them. If these bugs have open
>GitHub issues, be sure to tag them here as well, to keep the conversation
>linked together.

I'm sorry, what does this PR accomplish?

File: test/unit/store/toast-store.js [hunk 2]
```diff
@@ -153,4 +153,16 @@ describe('ToastStore', () => {
...
+    it('should add a toast when students are uploaded', () => {
+        Dispatcher.handleAction('STUDENTS_UPLOADED');
+        assert.lengthOf(ToastStore.getToasts(), 4);
+        assert.strictEqual(ToastStore.getToasts()[3].text, 'Students uploaded successfully');
+    });
+
+    it('should add a toast when students failed to uploaded', () => {
+        Dispatcher.handleAction('FILE_UNSUPPORTED');
+        assert.lengthOf(ToastStore.getToasts(), 4);
+        assert.strictEqual(ToastStore.getToasts()[3].text, 'Unknown file extension. File must be in excel format!');
```
Comment (mattmahn) [inline test/unit/store/toast-store.js]: Nit: Excel is a proper noun

**Comment by gearedmonkey** [comment]
@mattmahn Documentation added

File: src/views/components/student-file-upload.jsx [hunk 3]
```diff
@@ -1,24 +1,32 @@
 import React from 'react';
 import StudentFileUploadFormController from '../../controllers/components/student-file-upload-form';
-
```
Comment (thofmann) [inline src/views/components/student-file-upload.jsx]: Nit: Why?

File: src/views/components/student-file-upload.jsx [hunk 4]
```diff
@@ -1,24 +1,32 @@
...
-            <div className='file-upload-form'>
-              <form ref='uploadForm' onSubmit={this.submit.bind(this)}>
-                <input type='file' onChange={this.changeFile.bind(this)}></input>
-                <input type='submit'></input>
-              </form>
+        return(
+            <div>
+                <form onSubmit={this.submit.bind(this)}>
+                To upload a file:
+                    <input type="file" onChange={this.changeFile.bind(this)} accept=".xls,.xlsx, .xlsm" />
```
Comment (thofmann) [inline src/views/components/student-file-upload.jsx]: Nit: Inconsistent spacing after the commas in `accept`.

File: src/views/components/student-file-upload.jsx [hunk 5]
```diff
@@ -1,24 +1,32 @@
...
-        return (
-            <div className='file-upload-form'>
-              <form ref='uploadForm' onSubmit={this.submit.bind(this)}>
-                <input type='file' onChange={this.changeFile.bind(this)}></input>
-                <input type='submit'></input>
-              </form>
+        return(
+            <div>
+                <form onSubmit={this.submit.bind(this)}>
+                To upload a file:
```
Comment (thofmann) [inline src/views/components/student-file-upload.jsx]: Nit: Maybe something more descriptive here?

File: src/views/components/student-file-upload.jsx [hunk 6]
```diff
@@ -1,24 +1,32 @@
...
-
     render() {
-        return (
-            <div className='file-upload-form'>
-              <form ref='uploadForm' onSubmit={this.submit.bind(this)}>
-                <input type='file' onChange={this.changeFile.bind(this)}></input>
-                <input type='submit'></input>
-              </form>
+        return(
+            <div>
```
Comment (thofmann) [inline src/views/components/student-file-upload.jsx]: NACK: This page should have a title.

File: test/unit/lib/api-client.js [hunk 7]
```diff
@@ -315,11 +316,32 @@ describe('API Client', () => {
                 id: '123456'
             },
             response
-        })
+        });
```
Comment (thofmann) [inline test/unit/lib/api-client.js]: :clap:

File: test/unit/lib/api-client.js [hunk 8]
```diff
@@ -315,11 +316,32 @@ describe('API Client', () => {
...
         return searchStudent('123456').then(data => {
             assert.deepEqual(data, response.data);
             mockServer.validate();
         });
     });
 
+    it('uploadStudents', () => {
+        let response = {
+            status: 'success',
+            data: {
```
Comment (thofmann) [inline test/unit/lib/api-client.js]: NACK: This isn't a response that would be given by `POST /api/student`.

File: src/views/components/student-file-upload.jsx (hunk 6 above)
Comment (gearedmonkey) [inline src/views/components/student-file-upload.jsx]: Added a title and some text describing the requirements of the excel doc.

File: src/views/components/student-file-upload.jsx (hunk 5 above)
Comment (gearedmonkey) [inline src/views/components/student-file-upload.jsx]: Sure?

File: src/views/components/student-file-upload.jsx (hunk 4 above)
Comment (gearedmonkey) [inline src/views/components/student-file-upload.jsx]: fixed

File: test/unit/store/toast-store.js (hunk 2 above)
Comment (gearedmonkey) [inline test/unit/store/toast-store.js]: sure

File: test/unit/lib/api-client.js (hunk 8 above)
Comment (gearedmonkey) [inline test/unit/lib/api-client.js]: Fixed.

**Comment by thofmann** [comment]
Nit: Should `Mock_Students.xlsx` be somewhere in the `test` directory? It isn't documentation.

**Comment by thofmann** [comment]
Is there a reason we can't test `uploadStudents` and `validateFile` in `student-file-upload-form.js`?

**Pull request merged**